Sources/TuistCore/MetadataProviders/XCFrameworkSignatureProvider.swift

```diff
@@ -0,0 +1,94 @@
+import Command
+import CryptoKit
+import FileSystem
+import Foundation
+import Path
+import TSCBasic
+import TuistSupport
+import XcodeGraph
+
+enum XCFrameworkSignatureProviderError: FatalError, Equatable {
+    case codesignOutputMissing
+    case certificateFileReadFailed
+    case appleSignedXCFrameworkMissingDetails(teamIdentifier: String?, teamName: String?)
+
+    var description: String {
+        switch self {
+            case .codesignOutputMissing:
+            return "codesign finished, but no output file was found."
+            case .certificateFileReadFailed:
+                return "Failed to read certificate file."
+            case let .appleSignedXCFrameworkMissingDetails(teamIdentifier, teamName):
+            return "Apple signed XCFramework missing team identifier or name. teamIdentifier: \(teamIdentifier ?? "nil"), teamName: \(teamName ?? "nil")"
+        }
+    }
+    
+    var type: TuistSupport.ErrorType {
+        switch self {
+        case .codesignOutputMissing, .certificateFileReadFailed, .appleSignedXCFrameworkMissingDetails:
+            return .abort
+        }
+    }
+}
+
+/// A provider that can verify the signature calculated from an XCFramework.
+public struct XCFrameworkSignatureProvider {
+    private let commandRunner: CommandRunning
+    private let fileSystem: FileSysteming
+    private let codesignController: CodesignController
+
+    public init(
+        commandRunner: CommandRunning = CommandRunner(),
+        fileSystem: FileSysteming = FileSystem(),
+        codesignController: CodesignController = CodesignController()
+    ) {
+        self.commandRunner = commandRunner
+        self.fileSystem = fileSystem
+        self.codesignController = codesignController
+    }
+
+    private static let signedByAppleString = "Authority=Apple Root CA"
+    private static let teamNameRegExPattern = #"Authority=[^:]+?:\s*([^()]+)\s*\(([A-Z0-9]+)\)"#
+    private static let teamIdentifierRegExPattern = #"TeamIdentifier=([A-Z0-9]+)"#
+
+    /// Returns the signature of the XCFramework at the given `xcframeworkPath`.
+    public func signature(of xcframeworkPath: Path.AbsolutePath) async throws -> XCFrameworkSignature {
```

Comment:
I'd also cover this one with tests. In particular:

1. We get `.notSigned` if it's not signed
2. It returns it's signed by Apple if that's the case
3. It parses the signing metadata when signed with a different certificate.

Regarding 3., I can create one in the `fixtures/` directory signed with a Tuist certificate that we can use for testing purposes.